packages/cli/src/extensions.ts

```diff
@@ -5,20 +5,33 @@ import * as JSONSchemaFaker from 'json-schema-faker';
 import type { JSONSchemaFakerOptions } from 'json-schema-faker';
 import { resetJSONSchemaGenerator } from '@stoplight/prism-http';
 
-export async function configureExtensionsFromSpec(specFilePathOrObject: string | object): Promise<void> {
+export async function configureExtensionsUserProvided(
+  specFilePathOrObject: string | object,
+  cliParamOptions: { cliOption: string; cliValue: any }[]
+): Promise<void> {
   const result = decycle(await new $RefParser().dereference(specFilePathOrObject));
 
   resetJSONSchemaGenerator();
 
   forOwn(get(result, 'x-json-schema-faker', {}), (value: any, option: string) => {
-    if (option === 'locale') {
-      // necessary as workaround broken types in json-schema-faker
-      // @ts-ignore
-      return JSONSchemaFaker.locate('faker').setLocale(value);
+    setFakerValue(option, value);
+  });
+
+  // cli parameter takes precidence, so it is set after spec extensions are configed
+  cliParamOptions.forEach(param => {
+    if (param.cliValue !== undefined) {
+      setFakerValue(param.cliOption, param.cliValue);
     }
+  });
+}
 
+function setFakerValue(option: string, value: any) {
```

Comment:
I like the effect of factoring this out.  To me it seems like `configureExtensionsUserProvided()` is now at a more consistent level of detail/abstraction.